### PR TITLE
Backend reformatting (AlignmentRecordLayout)

### PR DESCRIPTION
**Pull request by matthew-zhao**
Removed references to track or TrackLayout in AlignmentRecordLayout. AlignmentRecordLayout now returns a ReadJson of AlignmentRecord information and list of MisMatchJsons (MatePairs not included).

### CODE REVIEW

File: mango-core/src/main/scala/org/bdgenomics/mango/layout/AlignmentRecordLayout.scala [hunk 1]
```diff
@@ -121,37 +110,24 @@ object MergedAlignmentRecordLayout extends Logging {
...
-  val readPairs: Map[String, List[(ReferenceRegion, CalculatedAlignmentRecord)]] = sequence.groupBy(_._2.record.getReadName)
-  addTracks
-  trackBuilder = trackBuilder.filter(_.records.nonEmpty)
+class AlignmentRecordLayout(values: Iterator[(ReferenceRegion, CalculatedAlignmentRecord)]) with Logging {
+  val sequence = values.toList
 
-  def addTracks {
-    readPairs.foreach {
+  def collect: Iterator[CalculatedAlignmentRecord] = {
+    val records = new ListBuffer[CalculatedAlignmentRecord]
```
Comment (akmorrow13) [inline mango-core/src/main/scala/org/bdgenomics/mango/layout/AlignmentRecordLayout.scala]: This shouldnt be agnostic to sample. Here, we loose what records correspond to what sample

File: mango-core/src/main/scala/org/bdgenomics/mango/layout/FeatureLayout.scala [hunk 2]
```diff
@@ -1,3 +1,4 @@
+//FILE TO BE DELETED
```
Comment (akmorrow13) [inline mango-core/src/main/scala/org/bdgenomics/mango/layout/FeatureLayout.scala]: You can remove these comments. We can worry about file deletion later.

File: mango-core/src/main/scala/org/bdgenomics/mango/layout/MismatchLayout.scala [hunk 3]
```diff
@@ -259,8 +258,8 @@ object PointMisMatch {
   }
 }
 
-// tracked MisMatch Json Object
-case class MisMatchJson(op: String, refCurr: Long, length: Long, sequence: String, refBase: String, track: Long)
+// MisMatch Json Object
+case class MisMatchJson(op: String, refCurr: Long, length: Long, sequence: String, refBase: String)
```
Comment (akmorrow13) [inline mango-core/src/main/scala/org/bdgenomics/mango/layout/MismatchLayout.scala]: We should keep mismatch, not mismatchjson. If you eliminated mismatchjson, just change the name here.

File: mango-core/src/main/scala/org/bdgenomics/mango/layout/AlignmentRecordLayout.scala [hunk 4]
```diff
@@ -167,12 +143,11 @@ object ReadJson {
...
    * An implementation of ReadJson which converts AlignmentRecord data to ReadJson
    *
-   * @param recs The list of (Reference, AlignmentRecord) tuples to lay out in json
-   * @param track js track number
+   * @param recs The list of (AlignmentRecord, List[MisMatchJson]) tuples to lay out in json
    * @return List of Read Json objects
    */
-  def apply(recs: List[(ReferenceRegion, AlignmentRecord)], track: Int): List[ReadJson] = {
-    recs.map(rec => new ReadJson(rec._2.getReadName, rec._2.getStart, rec._2.getEnd, rec._2.getReadNegativeStrand, rec._2.getSequence, rec._2.getCigar, rec._2.getMapq, track))
+  def apply(recs: List[(AlignmentRecord, List[MisMatchJson])]): List[ReadJson] = {
```
Comment (akmorrow13) [inline mango-core/src/main/scala/org/bdgenomics/mango/layout/AlignmentRecordLayout.scala]: Same as mismatch, let's get rid of the 'json' appending to all of these objects, as they dont contain visual-specific information

File: mango-core/src/main/scala/org/bdgenomics/mango/layout/AlignmentRecordLayout.scala (hunk 1 above)
Comment (matthew-zhao) [inline mango-core/src/main/scala/org/bdgenomics/mango/layout/AlignmentRecordLayout.scala]: Is that what this line of code does?
`sequence.groupBy(_._2.record.getReadName)`

Does that mean that I should be returning tuples of [String, CalculatedAlignmentRecord]?

Comment (akmorrow13) [inline mango-core/src/main/scala/org/bdgenomics/mango/layout/AlignmentRecordLayout.scala]: That actually groups by mate pairs (readName is the same for pairs) To group by sample, use getRecordGroupName from AlignmentRecord. This should be done before passed to the function

**Pull request closed without merging**